onnxscript/optimizer/_inliner.py

```diff
@@ -142,10 +147,11 @@ def clone_graph(self, graph: ir.Graph) -> ir.Graph:
         input_values = [self.clone_value(v) for v in graph.inputs]
         nodes = [self.clone_node(node) for node in graph]
         initializers = [self.clone_value(init) for init in graph.initializers.values()]
+        output_values = [self.clone_value(v) for v in graph.outputs]
 
         return ir.Graph(
             input_values,  # type: ignore
-            graph.outputs,
+            output_values,  # type: ignore
```

Comment:
Curious why # type: ignore is needed?

Reply:
Because of a limitation in the type annotation. Consider a function `f(x, y)`. We support/allow the actual-value of `y` to be omitted (an optional input) in a call to f. When we inline such a call, inside the body of the function, if we have a call to `someop(x, y)`, we will replace `y` by `None` (essentially, omitting the presumably optional input of someop). So, the dictionary used to do the substitution maps names to `ir.Value | None`. However, in some contexts, (such as the output value of the graph), we don't expect the value to be `None` (as per ONNX requirement).

I suppose it may be a good idea to actually check for this condition, and throw an appropriate exception. In a correct model, it shouldn't happen.